doc/rst/technotes/gpu.rst

```diff
@@ -0,0 +1,90 @@
+.. _readme-gpu:
+
+.. default-domain:: chpl
+
+=============================
+Support for GPU targeted code
+=============================
+
+Chapel includes preliminary work to target Nvidia GPUs using CUDA. This work is
+under active development and has not yet been tested under a wide variety of
+environments. We have tested it on systems with Nvidia Tesla P100 GPUs and CUDA
+11.0 and a system with Nvidia Tesla K20 GPUs with CUDA 10.2. The current
+implementation does not support multiple GPUs on one node and will only apply
+to certain `forall` loops.
+
+--------
+Overview
+--------
+
+To deploy code to a GPU embed the relevant code in an 'on' statement targeting
+the GPU locale (i.e. `here.getChild(1)`).
+
+Any arrays that are declared in the body of this `on` statement will be
+allocated on the GPU  Chapel will generate CUDA kernels for `forall` statements
+in the `on` block that are eligible.  Loops are eligible if they only make use
+of known compiler primitives that are fast and local are eligible for GPU code
+generation. Here "fast" means "safe to run in a signal handler" and "local"
+means "doesn't cause any network communication".
+
+Any non-eligible loop will be executed on the CPU.
+
+---------------------
+Setup and compilation
+---------------------
+
+To enable GPU support set the environmental variable: `CHPL_LOCALE_MODEL=gpu`
+before building Chapel.  Chapel's build system will automatically try and
+deduce where your installation of CUDA exists. If the build system fails to do
+this, or you would like to use a different CUDA installation set it using the
+`CHPL_CUDA_PATH` environmental variable.
+
+We also suggest setting `CHPL_RT_NUM_THREADS_PER_LOCALE=1` (this is necessary
+if using CUDA 10).
+
+To compile a program simply execute `chpl` as normal.  By default the generated
```

Comment:
I'll add this paragraph: 

In our current implementation of GPU support, we compile the generated kernels
into a `.fatbin` file that is loaded at runtime. We put this `.fatbin` file as
well as other temporary files produced during code generation into a
`_gpu_files` directory.  The name of this directory will be prefixed by the
executable name (if one is specified by passing it via `-o` or `--output` to
`chpl`), otherwise we will prefix it with the process ID for the Chapel
compiler. We intend to change this behavior in future releases.